Add trigger collisions.

moisesjpelaez opened the pull request:
Taking bits from [here](https://github.com/saharan/OimoPhysics/compare/master...Dev-MarkoF:OimoPhysics:master) from @Dev-MarkoF. One thing is manifolds should be cleared/detached in [World.hx](https://github.com/saharan/OimoPhysics/blob/3746033767741331fb576bf67f01f5178813a9f3/src/oimo/dynamics/World.hx) (but I'm not sure what does that mean).

I hope this helps as a starting point if it doesn't close #45 .

Tested and working in [Armory](https://github.com/armory3d/armory) 2024.07

saharan commented:
Thank you, it definitely helps!
I need to edit and test it before merge though, e.g. shapes should have the `isTrigger` flag, not rigid bodies because collision events occur per shape pair.

moisesjpelaez commented:
I moved the `isTrigger` flag to `Shape`. ~~I also attempted a fix when trigger collisions occur, by reducing the manifolds to 1 in `World.step()`, however after a trigger collision happens, sometimes the non-trigger body that collided pass through other non-trigger bodies for a few milliseconds.~~

Edit: I just fixed the trespassing issue.

moisesjpelaez commented:
Closed PR as I made the changes in my fork's main branch.
